client/my-sites/checkout/checkout/checkout.jsx

```diff
@@ -377,8 +381,46 @@ const Checkout = createReactClass( {
 		page( redirectPath );
 	},
 
+	hasTransferProduct() {
+		return cartItems.hasTransferProduct( this.props.cart );
+	},
+
+	setValidTransfer( domain ) {
```

Comment:
TIL that this works:
```
this.setState( {
    domainTransfers: Object.assign( {
        [ domain ]: true
    }, this.state.domainTransfers )
} );
```
Code untested (expect for the `[domain]` part, but you should get the idea :) (alternatively, use [`update`](https://github.com/kolodny/immutability-helper#computed-property-names)).